geetools/ee_image.py

```diff
@@ -1678,6 +1678,72 @@ def fromList(images: ee.List | list[ee.Image]) -> ee.Image:
         ic = ee.ImageCollection.fromImages(images)
         return ic.toBands().rename(bandNames)
 
+    def classToBands(self, class_info: dict, band: str | int = 0) -> ee.Image:
+        """Convert each class into a separate binary mask.
+
+        Args:
+            class_info: class information.
+            band: band that contains class information
+
+        Example:
+            .. code-block:: python
+
+                import ee, geetools
+
+                ee.Initialize()
+
+                class_info = {
+                  '2': 'dark',
+                  '3': 'shadow',
+                  '7': 'clouds_low',
+                  '8': 'clouds_mid',
+                  '9': 'clouds_high',
+                  '10': 'cirrus'
+                }
+                image = ee.Image('ee.Image("COPERNICUS/S2_SR_HARMONIZED/20230120T142709_20230120T143451_T18GYT")
+                decoded = image.geetools.classToBands(class_info, "SCL")
+        """
+        class_info = format_class_info(class_info)
+        # I don't see the class info coming from the server, so it'll client side until I get challenged
+        images = []
+        for class_value, class_name in class_info.items():
```

Comment:
would it be possible to convert the dictionary of the classes into a ee object and thus change a bit the logic with 2 `map` ? I still have the hope that one day they will open extensibility of the javascript interface so we can port all these cool stuff to the code editor (that's why I'm forcing myself to always use ee computed object in the parameters)

Reply:
Usually, I do this when the parameter can come from the server, for example, as it happens with the param `styleProperty` in the method [`ee.FeatureCollection.style`](https://developers.google.com/earth-engine/apidocs/ee-featurecollection-style). However, I find this confusing, because AFAIK you can use a `dict` as a property **only** when you compute it on the fly, but you cannot store a `dict` as a property when you save it as an GEE asset. If this is true (I tested it and seems to be true), to compute it on the fly you need to have it on the client-side, therefore I don't see the benefit of converting it to a server-side object and process it on the server.

Reply:
we could create a function that process/format it from [STAC](https://storage.googleapis.com/earthengine-stac/catalog/COPERNICUS/COPERNICUS_S2_SR_HARMONIZED.json), but it'll still be on the client-side. We can do it on a follow-up PR.

Reply:
definitely material for a follow-up PR.